docs/housing-counselor-data.md

```diff
@@ -0,0 +1,163 @@
+# Housing Counselor Data Processing
+
+The [Find a Housing Counselor tool](https://www.consumerfinance.gov/find-a-housing-counselor/) gets its data from the US Department of Housing and Urban Development. HUD provides an API to their list of approved counseling agencies.
+
+We generate the JSON data for the Find a Housing Counselor tool using the `hud_generate_json` management command. This page describes how that command works to transform HUD data to the JSON file format we use for the FAHC tool, including the following steps:
+  1. fetch agency listings from HUD
+  2. clean the results
+  3. fill in any missing latitude and longitude values
+  4. create files of the 10 nearest results for each US ZIP code
+
+
+## Fetch agency listings
+
+(_in `fetcher.py`_)
+
+Request every housing counselor in HUD's database with this request:
+
+```
+https://data.hud.gov/Housing_Counselor/searchByLocation?Lat=38.8951&Long=-77.0367&Distance=5000
+```
+
+It returns thousands of results like this:
+
+```json
+    {
+        "services": "DFC,FBC,PPC,RHC",
+        "languages": "ENG,SPA",
+        "agc_STATUS": "A",
+        "agc_SRC_CD": "HUD",
+        "counslg_METHOD": "Face to Face Counseling,Group Counseling,Internet Counseling,Phone Counseling",
+        "agcid": "80790",
+        "adr1": "24331 W. Eight Mile Road",
+        "adr2": " ",
+        "city": "DETROIT",
+        "email": "tmorris@abayomicdc.org",
+        "fax": "313-541-3820",
+        "nme": "ABAYOMI COMMUNITY DEVELOPMENT CORPORATION",
+        "phone1": "313-541-9828-222",
+        "statecd": "MI",
+        "weburl": "www.abayomicdc.org",
+        "zipcd": "48219-1028",
+        "agc_ADDR_LATITUDE": "42.442658",
+        "agc_ADDR_LONGITUDE": "-83.28329",
+        "parentid": "81228",
+        "county_NME": "",
+        "phone2": " ",
+        "mailingadr1": "24331 W. Eight Mile Road",
+        "mailingadr2": " ",
+        "mailingcity": "DETROIT",
+        "mailingzipcd": "48219-1028",
+        "mailingstatecd": "MI",
+        "state_NME": "Michigan",
+        "state_FIPS_CODE": null,
+        "faithbased": "Y",
+        "colonias_IND": "Y",
+        "migrantwkrs_IND": "N"
+    },
+```
+
+## Clean the results
+
+(_in `fetcher.py`_)
+
+We replace the 'languages' value with the full language names. We replace the 'services' value with fully spelled out service descriptions. Both of these mappings come from the HUD API.
+
+(_in `cleaner.py`_)
+
+Clean the data: Convert latitude and longitude values to Floats. Convert the the city and organization name to title case. Ensure email appears to be an email (otherwise leave it blank). Add `http://` to the URL if it is present.
```

Comment:
```suggestion
Clean the data: Convert latitude and longitude values to floats. Convert the the city and organization name to title case. Ensure email appears to be an email (otherwise leave it blank). Add `http://` to the URL if it is not already present.
```